bip-xxxx.mediawiki

```diff
@@ -0,0 +1,358 @@
+
+If the receiver does not support the version of the sender, they should send an error with the list of supported versions:
+<pre>
+{
+    "errorCode": "version-unsupported",
+    "supported" : [ 2, 3, 4 ],
+    "message": "The version is not supported anymore"
+}
+</pre>
+
+* <code>additionalfeeoutputindex=</code>, the preferred output from which to increase the fee for the added inputs. (default: <code>-1</code>)
+
+If the <code>additionalfeeoutputindex</code> is out of bounds or pointing to the payment output meant for the receiver, the receiver should ignore the parameter.
+Should be ignored in the [[#spare-change|spare change]] case.
+
+* <code>maxadditionalfeecontribution=</code>, an integer defining the maximum amount in satoshis that the sender is willing to contribute towards fees for the additional inputs. <code>maxadditionalfeecontribution</code> must be ignored if set to less than zero. (default: -1)
+
+Note that if <code>maxadditionalfeecontribution</code> is too low, the sender should create a transaction with RBF disabled, as the original transaction could replace the payjoin transaction.
+Should be ignored in the [[#spare-change|spare change]] case.
+
+* <code>minfeerate=</code>, a decimal in satoshi per vbyte that the sender can use to constraint the receiver to not drop the minimum fee rate too much.
+
+Should be ignored in the [[#spare-change|spare change]] case.
+
+==Rationale==
+
+There is several consequences of our proposal:
+
+* The receiver can bump the fee of the original transaction.
+* The receiver can modify the outputs of the original PSBT.
+* The sender must provide the UTXO information (Witness or previous transaction) in the PSBT.
+
+===Respecting the minimum relay fee policy===
+
+To be properly relayed, a Bitcoin transaction needs to pay at least 1 satoshi per virtual byte.
+When fees are low, the original transaction is already 1 satoshi per virtual byte, so if the receiver adds their own input, they need to make sure the fee is increased such that the rate does not drop below 1 satoshi per virtual byte.
+
+===Preventing mempool replacement===
+
+A safe way to implement payjoin, is for both the sender and receiver to try broadcasting the original transaction at some fixed interval period regardless of the state of the payjoin.
+
+If the receiver was not properly adding fees to the payjoin transaction, the original transaction would end up replacing the payjoin transaction in the mempool.
+
+===Defeating heuristics based on the fee calculation===
+
+Most wallets are creating a round fee rate (like 2 sat/b).
+If the payjoin transaction's fee was not increased by the added size, then those payjoin transactions could easily be identifiable on the blockchain.
+
+Not only would those transactions stand out by not having a round fee (like 1.87 sat/b), but any suspicion of payjoin could be confirmed by checking if removing one input would create a round fee rate.
+
+===Receiver does not need to be a full node===
```

Comment:
This is a bad thing. Can we fix it?

Reply:
No.  Any attempt at "fixing" it would only make the problem substantially worse. You could for instance omit information (e.g. what BIP79 does) that a full-node can easily access; but if you do that, and the person doesn't want to run a full-node they will get that information elsewhere (and that information is made available by a dozen  API providers). And now you are in a vastly worse situation.

This problem belongs in a separate discussion. But if you want people to run full nodes, you need to make it as easy and practical as possible. From talking to people, 90%+ of the reason people don't want to run a full-node is cause they don't want to go through the IBD and would like to jump-start with a snapshot of a pruned node.

And of course, you need to educate people on the benefits of running a full node. But gimmicks like making it harder for people who don't want to run a full node to need a full node will only backfire with unintended consequences.

Reply:
It's a way of incentivizing full nodes.

>if you want people to run full nodes, you need to make it as easy and practical as possible.

There is nothing that can be done to make it easier. The only alternative is to create bigger incentives.

Reply:
> This is a bad thing. Can we fix it?

Sadly, this would limit the users of this protocol to a ridiculous number.

Reply:
Only if Bitcoin has already failed anyway.